### DISCUSSION TITLE
Get the current extension directory path in package.json

### DISCUSSION THREAD

**Discussion by tbazant** [microsoft/vscode-discussions#620] [Extension Development QnA] [answered]
Hello! When developing a new extension, I include a `.vale.ini` configuration file in its root directory. I need to pint to that file by its absolute path from the `package.json` file. How can I obtain such path to the extension root directory on the file system? For example, based on the extension ID?

**Comment by gjsjohnmurray** [comment]
You can use `vscode.extensions.getExtension(id)` to get an [Extension&lt;T&gt;](https://code.visualstudio.com/api/references/vscode-api#Extension&lt;T&gt;) which has `extensionPath` as one of its properties.

**Comment by r3m0t** [comment, marked as answer]
This is available on the `ExtensionContext` object that is provided to your `activate` method. https://code.visualstudio.com/api/references/vscode-api#ExtensionContext

**Comment by tbazant** [comment]
Great that it's possible. But how exactly do i implement this in the `package.json` file? I'm a total newbie in VSCode API...and have no idea where to put it :-) Here is my current code `package.json` for completeness:
```
{
  "name": "suse-vale-styleguide",
  "displayName": "SUSE styleguide Vale rules",
  "description": "Vale rules for SUSE styleguide",
  "version": "0.0.1",
  "repository": "https://github.com/openSUSE/suse-vale-styleguide/",
  "publisher": "openSUSE",
  "engines": {
    "vscode": "^1.75.0"
  },
  "categories": [
    "Linters"
  ],
  "extensionDependencies": [
    "arrata-ai.vale-server"
  ],
  "contributes": {
    "configurationDefaults": {
      "vale.valeCLI.config": "/home/tbazant/devel/suse-vscode-doc/extensions/suse-vale-styleguide/.vale.ini"
    }
  }
}
```
